hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlValidator.java

```diff
@@ -295,9 +307,35 @@ public Void visit(SqlCall call) {
                 }
                 return super.visit(call);
             }
+
+            @Override
+            public Void visit(SqlIdentifier id) {
+                SqlValidatorTable table = getCatalogReader().getTable(id.names);
+                // not every identifier is a table
+                if (table != null) {
+                    HazelcastTable hazelcastTable = table.unwrap(HazelcastTable.class);
+                    if (hazelcastTable.getTarget() instanceof ViewTable) {
+                        String viewQuery = ((ViewTable) hazelcastTable.getTarget()).getViewQuery();
+                        QueryParser parser = new QueryParser(validator);
+                        try {
+                            QueryParseResult parseResult = parser.parse(viewQuery);
```

Comment:
We're doing double parsing here. There are also other issues in this code, so I created a follow-up issue #20155